File: 0205-isomorphic-strings/0205-isomorphic-strings.ts
function isIsomorphic(s: string, t: string): boolean {
    const sMatches: {} = {};
    const tMatches: {} = {};
    for (let i = 0; i < s.length; i ++) {
        if ((s[i] in sMatches && sMatches[s[i]] !== t[i]) || (t[i] in tMatches && tMatches[t[i]] !== s[i])) {
            return false;
        } else {
            sMatches[s[i]] = t[i];
            tMatches[t[i]] = s[i];
        };
    };
    return true;
};

     

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        map1 = {}
        map2 = {}

        for i in range(len(s)):
            if s[i] in map1:
                if map1[s[i]] != t[i]:
                    return False
            if t[i] in map2:
                if map2[t[i]] != s[i]:
                    return False

            map1[s[i]] = t[i]
            map2[t[i]] = s[i]

        return True
